src/views/SaveCredential/SaveCredential.tsx

```diff
@@ -1,52 +1,30 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useCallback, useState } from 'react';
 import { browser } from 'webextension-polyfill-ts';
 import { IAttestation } from '@kiltprotocol/types';
 
 import {
-  Credential,
   getCredentialDownload,
-  saveCredential,
   useIdentityCredentials,
+  usePendingCredentialCheck,
 } from '../../utilities/credentials/credentials';
 import { usePopupData } from '../../utilities/popups/usePopupData';
 
-import { saveChannel } from '../../channels/saveChannel/saveChannel';
-
 import { CredentialCard } from '../../components/CredentialCard/CredentialCard';
 
 import * as styles from './SaveCredential.module.css';
 
-function useSaveCredential(credential: Credential | undefined) {
-  useEffect(() => {
-    (async () => {
-      if (credential && credential.name && credential.name.length > 0) {
-        await saveCredential(credential);
-        await saveChannel.return(undefined);
```

Comment:
Could you please keep this line?

Reply:
I wasn't sure where I should move this line after the refactoring.

Isn't this already handled on the SKYC side? I didn't see any duplicate extra status messages after removing this line.

Reply:
Well, the SKYC already ignores errors coming from this dialog, but sending them still doesn’t seem right. I’d put that line anywhere in this component, and it doesn’t need a special hook for it.

```js
  useEffect(() => {
    (async () => {
      await saveChannel.return(undefined);
    })();
  }, []);

```